### PR TITLE
router: rename a badly named function

### PR DESCRIPTION
**Pull request by alyssawilk**

### CODE REVIEW

**Comment by alyssawilk** [comment]
(it's been bugging me that upstream filter encodeHeaders then does UpstreamFilterManager decodeHeaders.  Confusing!)

File: source/common/router/router.cc [hunk 1]
```diff
@@ -1602,7 +1602,7 @@ void Filter::onUpstreamComplete(UpstreamRequest& upstream_request) {
     }
   }
 
-  upstream_request.removeFromList(upstream_requests_);
+  callbacks_->dispatcher().deferredDelete(upstream_request.removeFromList(upstream_requests_));
```
Comment (snowp) [inline source/common/router/router.cc:1605]: Is this change to defer delete the request intentional? Seems unrelated to the name change

**Review by snowp — CHANGES_REQUESTED** [review]
Rename lgtm, one question

File: source/common/router/router.cc (hunk 1 above)
Comment (alyssawilk) [reply source/common/router/router.cc]: nope sorry, bad PR split thanks for catching

**Review by snowp — APPROVED** [review]
Thanks!
